scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java

```diff
@@ -644,79 +580,66 @@ public Double doInTransaction(Session session) {
 
     public double getMeanTaskRunningTime(String jobId) {
         final long id = Long.parseLong(jobId);
-        Double result = executeReadOnlyTransaction(new SessionWork<Double>() {
-            @Override
-            public Double doInTransaction(Session session) {
-                Query jobQuery = session.getNamedQuery("checkJobExistence").setParameter("id", id);
-                if (jobQuery.uniqueResult() == null) {
-                    return null;
-                }
-
-                Query query = session.getNamedQuery("getMeanTaskRunningTime").setParameter("id", id);
-
-                Double result = (Double) query.uniqueResult();
-                return result == null ? 0 : result;
+        Double result = executeReadOnlyTransaction(session -> {
+            Query jobQuery = session.getNamedQuery("checkJobExistence").setParameter("id", id);
+            if (jobQuery.uniqueResult() == null) {
+                return null;
             }
 
+            Query query = session.getNamedQuery("getMeanTaskRunningTime").setParameter("id", id);
+
+            Double result1 = (Double) query.uniqueResult();
+            return result1 == null ? 0 : result1;
         });
 
         return checkResult(id, result);
     }
 
     public int getTotalNumberOfHostsUsed(String jobId) {
         final long id = Long.parseLong(jobId);
-        int result = executeReadOnlyTransaction(new SessionWork<Integer>() {
-            @Override
-            public Integer doInTransaction(Session session) {
-                Query jobQuery = session.getNamedQuery("checkJobExistence").setParameter("id", id);
-                if (jobQuery.uniqueResult() == null) {
-                    return null;
-                }
-
-                Query query = session.getNamedQuery("getTotalNumberOfHostsUsed").setParameter("id", id);
-
-                return ((Long) query.uniqueResult()).intValue();
+        int result = executeReadOnlyTransaction(session -> {
+            Query jobQuery = session.getNamedQuery("checkJobExistence").setParameter("id", id);
+            if (jobQuery.uniqueResult() == null) {
+                return null;
             }
 
+            Query query = session.getNamedQuery("getTotalNumberOfHostsUsed").setParameter("id", id);
+
+            return ((Long) query.uniqueResult()).intValue();
         });
 
         return checkResult(id, result);
```

Comment:
checkResult sounds like a check true or false. But it seems to return the result after it has been checked? It is not clear what it checks and what it returns if the check failed.